src/charm.py

```diff
@@ -105,6 +110,22 @@ def _on_metrics_endpoint_relation_broken(self, event: RelationDepartedEvent):
         username = metrics_username(event.relation)
         self.control_socket.remove_metrics_user(username)
 
+    def _on_dbcluster_relation_changed(self, event):
+        ips = self.model.get_binding(event.relation).network.ingress_addresses
+
+        if len(ips) > 1:
+            logger.error('multiple possible DB bind addresses; set a dbcluster network binding')
+            self.unit.status = BlockedStatus(
+                'multiple possible DB bind addresses; set a dbcluster network binding')
+            return
+
+        ip = str(ips[0])
+        if self._db_bind_address == ip:
```

Comment:
this will ALWAYS be false.

the reason why your tests pass is https://github.com/canonical/operator/issues/736

you should consider writing your tests with [scenario](https://github.com/canonical/ops-scenario) :)